src/BootKeyboard/BootKeyboard.cpp

```diff
@@ -196,15 +196,29 @@ uint8_t BootKeyboard_::getLeds(void) {
 }
 
 uint8_t BootKeyboard_::getProtocol(void) {
-  return protocol;
+  if (boot_protocol_) {
+    return HID_BOOT_PROTOCOL;
+  } else {
+    return HID_REPORT_PROTOCOL;
+  }
 }
 
 void BootKeyboard_::setProtocol(uint8_t protocol) {
-  this->protocol = protocol;
+  if (protocol == HID_BOOT_PROTOCOL) {
+    boot_protocol_ = true;
+  } else {
+    boot_protocol_ = false;
+  }
```

Comment:
Instead of setting the `protocol` variable here, I leave it alone and set the new boolean `boot_protocol_`. The `protocol` variable stays set to `HID_REPORT_PROTOCOL`, set in the constructor:
```c++
BootKeyboard_::BootKeyboard_(void) : PluggableUSBModule(1, 1, epType), protocol(HID_REPORT_PROTOCOL), idle(1), leds(0) {
```